### ISSUE TITLE
Template styling broken?

### ISSUE THREAD

**Issue by aaron-mota** [vercel/ai-chatbot#613]
I just cloned the `main` branch and I can't get the styling to work?

Also, `Module '"react"' has no exported member 'useActionState'.` error.

<details>
<summary><b>package.json</b></summary>

```
 "dependencies": {
    "@ai-sdk/openai": "1.0.6",
    "@radix-ui/react-alert-dialog": "^1.1.2",
    "@radix-ui/react-dialog": "^1.1.2",
    "@radix-ui/react-dropdown-menu": "^2.1.2",
    "@radix-ui/react-icons": "^1.3.0",
    "@radix-ui/react-label": "^2.1.0",
    "@radix-ui/react-select": "^2.1.2",
    "@radix-ui/react-separator": "^1.1.0",
    "@radix-ui/react-slot": "^1.1.0",
    "@radix-ui/react-tooltip": "^1.1.3",
    "@radix-ui/react-visually-hidden": "^1.1.0",
    "@tailwindcss/forms": "^0.5.9",
    "@vercel/analytics": "^1.3.1",
    "@vercel/blob": "^0.24.1",
    "@vercel/postgres": "^0.10.0",
    "ai": "4.0.10",
    "bcrypt-ts": "^5.0.2",
    "class-variance-authority": "^0.7.0",
    "classnames": "^2.5.1",
    "clsx": "^2.1.1",
    "date-fns": "^4.1.0",
    "diff-match-patch": "^1.0.5",
    "dotenv": "^16.4.5",
    "drizzle-orm": "^0.34.0",
    "fast-deep-equal": "^3.1.3",
    "framer-motion": "^11.3.19",
    "geist": "^1.3.1",
    "lucide-react": "^0.446.0",
    "nanoid": "^5.0.8",
    "next": "15.0.3-canary.2",
    "next-auth": "5.0.0-beta.25",
    "next-themes": "^0.3.0",
    "orderedmap": "^2.1.1",
    "postcss-nesting": "^13.0.1",
    "postgres": "^3.4.4",
    "prosemirror-example-setup": "^1.2.3",
    "prosemirror-inputrules": "^1.4.0",
    "prosemirror-markdown": "^1.13.1",
    "prosemirror-model": "^1.23.0",
    "prosemirror-schema-basic": "^1.2.3",
    "prosemirror-schema-list": "^1.4.1",
    "prosemirror-state": "^1.4.3",
    "prosemirror-view": "^1.34.3",
    "react": "19.0.0-rc-45804af1-20241021",
    "react-dom": "19.0.0-rc-45804af1-20241021",
    "react-markdown": "^9.0.1",
    "remark-gfm": "^4.0.0",
    "server-only": "^0.0.1",
    "sonner": "^1.5.0",
    "swr": "^2.2.5",
    "tailwind-merge": "^2.5.2",
    "tailwindcss-animate": "^1.0.7",
    "usehooks-ts": "^3.1.0",
    "zod": "^3.23.8"
  },
  "devDependencies": {
    "@biomejs/biome": "1.9.4",
    "@tailwindcss/typography": "^0.5.15",
    "@types/d3-scale": "^4.0.8",
    "@types/node": "^22.8.6",
    "@types/pdf-parse": "^1.1.4",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "drizzle-kit": "^0.25.0",
    "eslint": "^8.57.0",
    "eslint-config-next": "14.2.5",
    "eslint-config-prettier": "^9.1.0",
    "eslint-import-resolver-typescript": "^3.6.3",
    "eslint-plugin-tailwindcss": "^3.17.5",
    "postcss": "^8",
    "tailwindcss": "^3.4.1",
    "tsx": "^4.19.1",
    "typescript": "^5.6.3"
  }
  ```

</details>

![CleanShot 2024-12-11 at 21 35 58@2x](https://github.com/user-attachments/assets/33866b10-9f0c-4947-b075-dc92ff532581)

**Comment by aaron-mota** [comment]
Removing `--turbo` made the styling work (`"dev": "next dev --turbo"` to `"dev": "next dev"`)

**Comment by aaron-mota** [comment]
My react `@types/react` & `@types/react-dom` were behind (need to be 18.3+), that solved the `useActionState` issue

**Comment by aaron-mota** [comment]
After updating package versions, I could add the `--turbo` back to the `dev` command -- not sure how everything got so messed up.

**Comment by metadimensions** [comment]
still after updating the package versions I tried adding the turbo to the dev command but its still throws the styling error...

**Comment by metadimensions** [comment]
Hey @aaron-mota after updating the package versions... i tried adding the --turbo to the dev command but still the styling issue is there... without turbo there is no styling issue....

**Comment by aaron-mota** [comment]
> Hey @aaron-mota after updating the package versions... i tried adding the --turbo to the dev command but still the styling issue is there... without turbo there is no styling issue....

Here's my full package.json.  I'd make sure your react, next, and tailwindcss package versions look similar (e.g. NextJS v15+, React v19+, ...).

<details>
<summary><b>package.json</b></summary>

{
  "name": "ai-chatbot",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev --turbo",
    "build": "tsx lib/db/migrate && next build",
    "start": "next start",
    "lint": "next lint && biome lint --write --unsafe",
    "lint:fix": "next lint --fix && biome lint --write --unsafe",
    "format": "biome format --write",
    "db:generate": "drizzle-kit generate",
    "db:migrate": "npx tsx lib/db/migrate.ts",
    "db:studio": "drizzle-kit studio",
    "db:push": "drizzle-kit push",
    "db:pull": "drizzle-kit pull",
    "db:check": "drizzle-kit check",
    "db:up": "drizzle-kit up"
  },
  "dependencies": {
    "@ai-sdk/openai": "1.0.6",
    "@radix-ui/react-alert-dialog": "^1.1.2",
    "@radix-ui/react-dialog": "^1.1.2",
    "@radix-ui/react-dropdown-menu": "^2.1.2",
    "@radix-ui/react-icons": "^1.3.0",
    "@radix-ui/react-label": "^2.1.0",
    "@radix-ui/react-select": "^2.1.2",
    "@radix-ui/react-separator": "^1.1.0",
    "@radix-ui/react-slot": "^1.1.0",
    "@radix-ui/react-tooltip": "^1.1.3",
    "@radix-ui/react-visually-hidden": "^1.1.0",
    "@vercel/analytics": "^1.3.1",
    "@vercel/blob": "^0.24.1",
    "@vercel/postgres": "^0.10.0",
    "ai": "4.0.10",
    "bcrypt-ts": "^5.0.2",
    "class-variance-authority": "^0.7.0",
    "classnames": "^2.5.1",
    "clsx": "^2.1.1",
    "date-fns": "^4.1.0",
    "diff-match-patch": "^1.0.5",
    "dotenv": "^16.4.5",
    "drizzle-orm": "^0.34.0",
    "fast-deep-equal": "^3.1.3",
    "framer-motion": "^11.3.19",
    "geist": "^1.3.1",
    "lucide-react": "^0.446.0",
    "nanoid": "^5.0.8",
    "next": "15.0.3-canary.2",
    "next-auth": "5.0.0-beta.25",
    "next-themes": "^0.3.0",
    "orderedmap": "^2.1.1",
    "postgres": "^3.4.4",
    "prosemirror-example-setup": "^1.2.3",
    "prosemirror-inputrules": "^1.4.0",
    "prosemirror-markdown": "^1.13.1",
    "prosemirror-model": "^1.23.0",
    "prosemirror-schema-basic": "^1.2.3",
    "prosemirror-schema-list": "^1.4.1",
    "prosemirror-state": "^1.4.3",
    "prosemirror-view": "^1.34.3",
    "react": "19.0.0-rc-45804af1-20241021",
    "react-dom": "19.0.0-rc-45804af1-20241021",
    "react-markdown": "^9.0.1",
    "remark-gfm": "^4.0.0",
    "server-only": "^0.0.1",
    "sonner": "^1.5.0",
    "swr": "^2.2.5",
    "tailwind-merge": "^2.5.2",
    "tailwindcss-animate": "^1.0.7",
    "usehooks-ts": "^3.1.0",
    "zod": "^3.23.8"
  },
  "devDependencies": {
    "@biomejs/biome": "1.9.4",
    "@tailwindcss/typography": "^0.5.15",
    "@types/d3-scale": "^4.0.8",
    "@types/node": "^22.8.6",
    "@types/pdf-parse": "^1.1.4",
    "@types/react": "^18.3.1",
    "@types/react-dom": "^18.3.1",
    "drizzle-kit": "^0.25.0",
    "eslint": "^8.57.0",
    "eslint-config-next": "14.2.5",
    "eslint-config-prettier": "^9.1.0",
    "eslint-import-resolver-typescript": "^3.6.3",
    "eslint-plugin-tailwindcss": "^3.17.5",
    "postcss": "^8",
    "tailwindcss": "^3.4.1",
    "tsx": "^4.19.1",
    "typescript": "^5.6.3"
  }
}

</details>

Also, try to remove any cached stuff:
`rm -rf node_modules .next`
`pnpm install`
`pnpm build`

If that doesn't work, maybe you need to remove the pnpm-lock.yaml file too (if using pnpm, otherwise whichever lock file you have for your package manager):
`rm -rf node_modules .next`
`rm pnpm-lock.yaml`
`pnpm store prune
pnpm store clean` consider this if using pnpm
`pnpm install`
`pnpm build`

Try fully exiting out of the browser and restarting it too, rather than relying on a hot reload to fix it.

**Comment by metadimensions** [comment]
@aaron-mota 

Yes I tried its working fine with "dev": "next dev " but if I add  "dev": "next dev --turbo" the UI loads like this : 
<img width="515" alt="Screenshot 2024-12-27 at 11 11 59 PM" src="https://github.com/user-attachments/assets/8f3a4938-1b9a-4db3-af73-4d922997a3d5" />

**Comment by aaron-mota** [comment]
@metadimensions 

Can you share your package.json?

Also, just want to make sure you tried everything from my previous message?

**Comment by metadimensions** [comment]
@aaron-mota  yes i have tried the above steps the code is up and running without `turbo` command in package.json which looks like this

```
{
  "name": "ai-chatbot",
  "version": "0.1.0",
  "private": true,
  "scripts": {
  "dev": "next dev --turbo",
  "build": "tsx lib/db/migrate && next build",
  "start": "next start",
  "lint": "next lint && biome lint --write --unsafe",
  "lint:fix": "next lint --fix && biome lint --write --unsafe",
  "format": "biome format --write",
  "db:generate": "drizzle-kit generate",
  "db:migrate": "npx tsx lib/db/migrate.ts",
  "db:studio": "drizzle-kit studio",
  "db:push": "drizzle-kit push",
  "db:pull": "drizzle-kit pull",
  "db:check": "drizzle-kit check",
  "db:up": "drizzle-kit up"
  },
  "dependencies": {
  "@ai-sdk/openai": "1.0.6",
  "@radix-ui/react-alert-dialog": "^1.1.2",
  "@radix-ui/react-dialog": "^1.1.2",
  "@radix-ui/react-dropdown-menu": "^2.1.2",
  "@radix-ui/react-icons": "^1.3.0",
  "@radix-ui/react-label": "^2.1.0",
  "@radix-ui/react-select": "^2.1.2",
  "@radix-ui/react-separator": "^1.1.0",
  "@radix-ui/react-slot": "^1.1.0",
  "@radix-ui/react-tooltip": "^1.1.3",
  "@radix-ui/react-visually-hidden": "^1.1.0",
  "@vercel/analytics": "^1.3.1",
  "@vercel/blob": "^0.24.1",
  "@vercel/postgres": "^0.10.0",
  "ai": "4.0.10",
  "bcrypt-ts": "^5.0.2",
  "class-variance-authority": "^0.7.0",
  "classnames": "^2.5.1",
  "clsx": "^2.1.1",
  "date-fns": "^4.1.0",
  "diff-match-patch": "^1.0.5",
  "dotenv": "^16.4.5",
  "drizzle-orm": "^0.34.0",
  "fast-deep-equal": "^3.1.3",
  "framer-motion": "^11.3.19",
  "geist": "^1.3.1",
  "lucide-react": "^0.446.0",
  "nanoid": "^5.0.8",
  "next": "15.0.3-canary.2",
  "next-auth": "5.0.0-beta.25",
  "next-themes": "^0.3.0",
  "orderedmap": "^2.1.1",
  "postgres": "^3.4.4",
  "prosemirror-example-setup": "^1.2.3",
  "prosemirror-inputrules": "^1.4.0",
  "prosemirror-markdown": "^1.13.1",
  "prosemirror-model": "^1.23.0",
  "prosemirror-schema-basic": "^1.2.3",
  "prosemirror-schema-list": "^1.4.1",
  "prosemirror-state": "^1.4.3",
  "prosemirror-view": "^1.34.3",
  "react": "19.0.0-rc-45804af1-20241021",
  "react-dom": "19.0.0-rc-45804af1-20241021",
  "react-markdown": "^9.0.1",
  "remark-gfm": "^4.0.0",
  "server-only": "^0.0.1",
  "sonner": "^1.5.0",
  "swr": "^2.2.5",
  "tailwind-merge": "^2.5.2",
  "tailwindcss-animate": "^1.0.7",
  "usehooks-ts": "^3.1.0",
  "zod": "^3.23.8"
  },
  "devDependencies": {
  "@biomejs/biome": "1.9.4",
  "@tailwindcss/typography": "^0.5.15",
  "@types/d3-scale": "^4.0.8",
  "@types/node": "^22.8.6",
  "@types/pdf-parse": "^1.1.4",
  "@types/react": "^18.3.1",
  "@types/react-dom": "^18.3.1",
  "drizzle-kit": "^0.25.0",
  "eslint": "^8.57.0",
  "eslint-config-next": "14.2.5",
  "eslint-config-prettier": "^9.1.0",
  "eslint-import-resolver-typescript": "^3.6.3",
  "eslint-plugin-tailwindcss": "^3.17.5",
  "postcss": "^8",
  "tailwindcss": "^3.4.1",
  "tsx": "^4.19.1",
  "typescript": "^5.6.3"
  }
  }
```

**Comment by aaron-mota** [comment]
Hmm, I'm really not sure then, sorry.

**Comment by Garfield-yin** [comment]
I have the same question

**Comment by aaron-mota** [comment]
@Garfield-yin if you've tried everything above, I'm not sure how to help, unfortunately.  One thing I would try is, if the styling is working without "--turbo", and you've done all the things I've mentioned above, try adding the "--turbo" back in and then opening localhost on a private/incognito browser (which will ensure there's no caching of the broken styling going on).  If that still doesn't work, I'm not sure :(
